pkg/virt-handler/rest/lifecycle.go

```diff
@@ -272,67 +182,61 @@ func (lh *LifecycleHandler) GetGuestInfo(request *restful.Request, response *res
 }
 
 func (lh *LifecycleHandler) GetUsers(request *restful.Request, response *restful.Response) {
-	vmi, code, err := getVMI(request, lh.vmiInformer)
+	vmi, client, err := lh.getVMILauncherClient(request, response)
 	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error(failedRetrieveVMI)
-		response.WriteError(code, err)
 		return
 	}
 
 	log.Log.Object(vmi).Infof("Retreiving userlist from %s", vmi.Name)
 
-	sockFile, err := cmdclient.FindSocketOnHost(vmi)
+	userList, err := client.GetUsers()
 	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error(failedDetectCmdClient)
+		log.Log.Object(vmi).Reason(err).Error("Failed to get user list")
 		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
-	client, err := cmdclient.NewClient(sockFile)
+
+	response.WriteEntity(userList)
+}
+
+func (lh *LifecycleHandler) GetFilesystems(request *restful.Request, response *restful.Response) {
+	vmi, client, err := lh.getVMILauncherClient(request, response)
 	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error(failedConnectCmdClient)
-		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
 
-	userList, err := client.GetUsers()
+	log.Log.Object(vmi).Infof("Retreiving filesystem list from %s", vmi.Name)
+
+	fsList, err := client.GetFilesystems()
 	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error("Failed to get user list")
+		log.Log.Object(vmi).Reason(err).Error("Failed to get guest info")
 		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
 
-	response.WriteEntity(userList)
+	response.WriteEntity(fsList)
 }
 
-func (lh *LifecycleHandler) GetFilesystems(request *restful.Request, response *restful.Response) {
+func (lh *LifecycleHandler) getVMILauncherClient(request *restful.Request, response *restful.Response) (*v1.VirtualMachineInstance, cmdclient.LauncherClient, error) {
```

Comment:
It seems that this method is doing two things: Revealing the VMI and the Client.
Can we separate these two so each will do one thing?

Also, it seems that in case of error, the same action is taken. So can we return the error and "code" so the caller will perform the `WriteError`? It is a bit odd to see the returned error being ignored by the callers in most cases.

Reply:
> I am a bit worried that this area of code is not unit tested at all.
Are you sure all the changes here are covered any some test?

I didn't really look into the unit tests or functional tests for this. There is the `tests/vmi_lifecycle_test.go` that I ran successfully locally before this PR, but I am unsure of what exactly is being covered there.

Let me know where coverage is lacking, and I can adress it in another PR.

Reply:
@EdDev Would you like to follow up?

Reply:
I think this change is a great step forward, I do not want to block it on things I see can get even better.
Sorry for not responding for so long.